### DISCUSSION TITLE
Working example with CRA/craco

### DISCUSSION THREAD

**Discussion by amorriscode** [vanilla-extract-css/vanilla-extract#358] [Q&A] [answered]
Hey all! 👋

Love the project.

I was just wondering, has anyone successfully used `vanilla-extract` with `create-react-app`/`craco`? I've tried to get it working using the basic instructions but `mini-css-extract-plugin` v2 isn't compatible (`craco` is using webpack 4). It's the same error as https://github.com/seek-oss/vanilla-extract/issues/296.

When I downgrade `mini-css-extract-plugin`, I also get errors. 😅 

```bash
Syntax error: Unknown word (4:7)

  2 | export {};
  3 |     if(module.hot) {
> 4 |       // 1631583113062
    |       ^
  5 |       var cssReload = require("/Users/amorriscode/dev/monthly/frontend/node_modules/mini-css-extract-plugin/dist/hmr/hotModuleReplacement.js")(module.id, {"locals":false});
  6 |       module.hot.dispose(cssReload);
```

There's also a [codesandbox](https://codesandbox.io/s/craco-vanilla-extract-boiler-plate-q71d9) which is close but still doesn't work. 😄 

So, who is using Vanilla Extract with CRA? What'd you have to do to make it happy?

## Update

I may have had it working this whole time.

I thought that it wasn't working properly because, upon inspection, this is where my CSS seemed to come from on the production build (`vanilla-extract-webpack-plugin-extracted.cjs.js`):

![image](https://user-images.githubusercontent.com/16005567/133307864-c8394b22-f6d6-4e2b-8ff4-4bd90cbfd9c3.png)

However, I cloned the [official Sandbox](https://codesandbox.io/s/github/seek-oss/vanilla-extract/tree/master/examples/webpack-react?file=/src/App.css.ts) to my computer and realized the production build produces the same result. I suppose this is expected behaviour then? 😅

**Comment by mattcompiles** [comment, marked as answer]
Yeah it's a bit strange. It's because webpack has no concept of virtual files and we use a technique called `matchResource`. However ,this causes all CSS files to seem like they're coming from a single file, which in reality they aren't.

**Reply by amorriscode** [reply]
Gotcha, thanks @mattcompiles. Sorry for the noise!
